src/common/components/ui/Form/Form.tsx

```diff
@@ -0,0 +1,66 @@
+import React, { ReactElement } from 'react';
+import { SubmitHandler, useForm, UseFormProps, FieldErrors } from 'react-hook-form';
+
+type props<T> = {
+  form?: UseFormProps<T>;
+  children: ReactElement | ReactElement[];
+  onSubmit: SubmitHandler<T>;
+  // schema?: any;
+};
+
+// const useYupValidationResolver = (validationSchema) =>
+//   useCallback(
+//     async (data) => {
+//       try {
+//         const values = await validationSchema.validate(data, {
+//           abortEarly: false,
+//         });
+
+//         return {
+//           values,
+//           errors: {},
+//         };
+//       } catch (errors) {
+//         return {
+//           values: {},
+//           errors: errors.inner.reduce(
+//             (allErrors, currentError) => ({
+//               ...allErrors,
+//               [currentError.path]: {
+//                 type: currentError.type ?? 'validation',
+//                 message: currentError.message,
+//               },
+//             }),
+//             {}
+//           ),
+//         };
+//       }
+//     },
+//     [validationSchema]
+//   );
+
+export default <T,>({ form, children, onSubmit }: props<T>) => {
+  // const resolver = useYupValidationResolver(schema);
+  const methods = useForm<T>(form);
+  const {
+    handleSubmit,
+    formState: { errors },
+  } = methods;
+
+  return (
+    <form onSubmit={handleSubmit(onSubmit)}>
+      {React.Children.map(children, (child) =>
+        child.props.name
+          ? React.createElement(child.type, {
+              ...{
```

Comment:
I'd suggest to not use spread operator on props since this tend to make the code very hard to read (we don't exactly know what type of props we are passing in, and it will make the reader to inspect the code to figure it out).
If this is an edge case, you can deactivate this behavior only on this file and let the rest of the app to validate this eslint rule